index.js

```diff
@@ -44,12 +44,31 @@ var HOST_PORT_PATTERN = /^(\d+.\d+.\d+.\d+):\d+$/;
 var MAX_JOIN_DURATION = 300000;
 var PROXY_REQ_PROPS = ['keys', 'dest', 'req', 'res'];
 
+var ArgumentRequiredError = TypedError({
```

Comment:
yep, agreed. i feel we're getting to that point where it'd be useful to have such a file.